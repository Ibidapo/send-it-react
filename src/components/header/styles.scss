@import '../../styles/base/settings';

.main-header {
  position: relative;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-items: center;
  background-color: white;
  padding: 0 $l-size;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.3);
  z-index: 1;
  transition: all ease-in-out 400ms;
}

.dashboard {
  background: transparent;
  box-shadow: none;
}

.main-header {
  height: 90px;
}

#list > a {
  padding: 0.75rem;
  border-radius: 0.5rem;
  text-decoration: none;
  margin: 0 1rem;
}

.main-header > #list > a:hover {
  background-color: whitesmoke;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
}

#menu {
  background: transparent;
  display: none;
  cursor: pointer;
  border: none;
  outline: none;
}
#menu > .bar1, #menu > .bar2, #menu > .bar3 {
  width: 35px;
  height: 5px;
  background-color: #333;
  margin: 6px 0;
  transition: 0.4s;
}
#menu.active > .bar1 {
  -webkit-transform: rotate(-45deg) translate(-9px, 6px);
  transform: rotate(-45deg) translate(-9px, 6px);
}
#menu.active > .bar2 {
  opacity: 0;
}
#menu.active > .bar3 {
  -webkit-transform: rotate(45deg) translate(-8px, -8px);
  transform: rotate(45deg) translate(-8px, -8px);
}

/* Styling for Tablet and Mobile devices */
@media screen and (max-width: 1024px) {
  .main-header {
    height: auto;

    > .d-flex {
      height: 90px;
      align-items: center;
      width: 100%;
    }

    #menu {
      display: block;
    }

    #list {
      display: none;
    }

    #list.active {
      display: flex;
      flex-direction: column;
      padding: 0 0 1rem;
    }

    #list > a {
      margin: 0.5rem 0;
    }
  }

  .dashboard {
    padding: 0 1rem;
  }
}