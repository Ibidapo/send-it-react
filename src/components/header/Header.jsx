import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { logoutUser } from '../../redux/actions/auth';
import { clearProfile } from '../../redux/actions/profile';
import './styles.scss';

class Header extends Component {
  state = {
    toggle: false
  }

  toggleHeader = () => this.setState((prevState) => ({ toggle: !prevState.toggle }));
  handleLogout = () => {
    const { props: { logoutUser, clearProfile } } = this;
    clearProfile();
    logoutUser();
  }

  render() {
    const { state: { toggle }, props: { isLoggedIn }, toggleHeader, handleLogout } = this;

    return (
      <header className="main-header">
        <div className="d-flex space-between">
          <button 
            id="menu"  
            onClick={toggleHeader}
            className={toggle ? 'active' : ''}
          >
            <div className="bar1"></div>
            <div className="bar2"></div>
            <div className="bar3"></div>
          </button>
          <Link to='/'>
            <img src="images/logo.png" alt="SendIT-logo" className="logo" />
          </Link>
        </div>
        <div id="list" className={toggle ? 'active' : ''}>
          {
            isLoggedIn ?
            (
              <React.Fragment>
                <Link to='/' onClick={handleLogout}>Logout</Link>
                <Link to="#contact" className="purple-btn">My Dashboard</Link>
              </React.Fragment>
            ) : (
              <React.Fragment>
                <Link to="#about">About</Link>
                <Link to="#contact">Contact</Link>
                <Link to="#login" className="purple-btn">Login / Register</Link>
              </React.Fragment>
            )
          }
        </div>
      </header>
    );
  }
}

Header.propTypes = {
  isLoggedIn: PropTypes.bool,
  logoutUser: PropTypes.func,
  clearProfile: PropTypes.func
}

const mapStateToProps = ({ auth: { isLoggedIn } }) => ({ isLoggedIn });
const mapDispatchToProps = ({ logoutUser, clearProfile })

export default connect(mapStateToProps, mapDispatchToProps)(Header);