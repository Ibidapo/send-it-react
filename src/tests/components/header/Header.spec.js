import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../../../components/header/Header';

let wrapper;

describe('Test for Header Component', () => {
  it ('Should render correctly', () => {
    wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  })

  it('Should contain, when not logged out', () => {
    const props = { isLoggedIn: false }
    wrapper = shallow(<Header {...props} />);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('Link').length).toBe(4);
    expect(wrapper.find('div').length).toBe(5)
    wrapper.find('button').at(0).simulate('click');
  })

  it('Should contain, when logged in', () => {
    const props = { isLoggedIn: true, logoutUser: jest.fn() , clearProfile: jest.fn() }
    wrapper = shallow(<Header {...props} />);
    expect(wrapper.find('Link').length).toBe(1);
    wrapper.find('Link').at(0).simulate('click');
  })
});