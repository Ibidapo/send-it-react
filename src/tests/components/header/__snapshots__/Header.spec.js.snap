// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header Component Should render correctly 1`] = `
<header
  className="main-header "
>
  <div
    className="d-flex space-between"
  >
    <button
      className=""
      id="menu"
      onClick={[Function]}
    >
      <div
        className="bar1"
      />
      <div
        className="bar2"
      />
      <div
        className="bar3"
      />
    </button>
    <Link
      to="/"
    >
      <img
        alt="SendIT-logo"
        className="logo"
        src="images/logo.png"
      />
    </Link>
  </div>
  <div
    className=""
    id="list"
  >
    <Link
      to="#about"
    >
      About
    </Link>
    <Link
      to="#contact"
    >
      Contact
    </Link>
    <Link
      className="purple-btn"
      to="#login"
    >
      Login / Register
    </Link>
  </div>
</header>
`;
